src/parser/jobs/drk/modules/Delirium.tsx

```diff
@@ -0,0 +1,187 @@
+import {t} from '@lingui/macro'
+import {Plural, Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {RotationTable} from 'components/ui/RotationTable'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import {BuffEvent, CastEvent} from 'fflogs'
+import _ from 'lodash'
+import Module, {dependency} from 'parser/core/Module'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+import React from 'react'
+
+const DELIRIUM_DURATION = 10000
+
+const BLOOD_SKILLS = [
+	ACTIONS.BLOODSPILLER.id,
+	ACTIONS.QUIETUS.id, // obvious caveat - only on >1 target
+]
+
+const EXPECTED_CONSTANTS = {
+	GCD: 5,
+	BLOOD_SKILLS: 5,
+}
+
+const SEVERITY_MISSED_GCDS = {
+	1: SEVERITY.MEDIUM,
+	3: SEVERITY.MAJOR,
+}
+
+const SEVERITY_MISSED_BLOOD_SKILLS = {
+	1: SEVERITY.MEDIUM,
+	2: SEVERITY.MAJOR,
+}
+
+class DeliriumState {
+	start: number
+	end: number | null = null
```

Comment:
Generally prefer `foo?: type` over `foo: type | null = null`. There's a reasonable number of nulls in use throughout the codebase, but I'm trying to slowly normalise them down to `undefined` over time.

Reply:
Adjusted.  Though the way I have PhpStorm configured is not a fan of that syntax. 🤣